datafusion/core/src/datasource/file_format/csv.rs

```diff
@@ -65,6 +65,59 @@ impl Default for CsvFormat {
 }
 
 impl CsvFormat {
+    /// Return a newline delimited stream from the specified file on
+    /// Stream, decompressing if necessary
+    /// Each returned `Bytes` has a whole number of newline delimited rows
+    async fn read_to_delimited_chunks(
```

Comment:
I feel like there must be a simpler way to express this code, but this does appear to work.

I wonder if we could use `BoxStream` rather than `impl Stream....` 🤔

Reply:
fixed,thanks a lot.